x/ibc/07-tendermint/types/misbehaviour_test.go

```diff
@@ -300,26 +280,52 @@ func (suite *TendermintTestSuite) TestCheckMisbehaviourAndUpdateState() {
 				ChainID:  chainID,
 				ClientID: chainID,
 			},
-			simapp.DefaultConsensusParams,
 			suite.now,
 			false,
 		},
 	}
 
 	for i, tc := range testCases {
 		tc := tc
+		suite.Run(fmt.Sprintf("Case: %s", tc.name), func() {
+
+			if i != 10 {
+				return
+			}
+
+			// reset suite to create fresh application state
+			suite.SetupTest()
+
+			// Set current timestamp in context
+			ctx := suite.chainA.GetContext().WithBlockTime(tc.timestamp)
+			ctx = ctx.WithConsensusParams(simapp.DefaultConsensusParams)
+
+			// Set trusted consensus states in client store
+
+			if tc.consensusState1 != nil {
+				suite.chainA.App.IBCKeeper.ClientKeeper.SetClientConsensusState(ctx, clientID, tc.consensusState1.GetHeight(), tc.consensusState1)
+			}
+			if tc.consensusState2 != nil {
+				suite.chainA.App.IBCKeeper.ClientKeeper.SetClientConsensusState(ctx, clientID, tc.consensusState2.GetHeight(), tc.consensusState2)
+			}
 
-		clientState, err := tendermint.CheckMisbehaviourAndUpdateState(tc.clientState, tc.consensusState1, tc.consensusState2, tc.evidence, tc.timestamp, tc.consensusParams)
+			clientState, err := tc.clientState.CheckMisbehaviourAndUpdateState(
+				ctx,
+				suite.cdc,
+				suite.chainA.App.IBCKeeper.ClientKeeper.ClientStore(ctx, clientID), // pass in clientID prefixed clientStore
+				tc.evidence,
+			)
 
-		if tc.expPass {
-			suite.Require().NoError(err, "valid test case %d failed: %s", i, tc.name)
-			suite.Require().NotNil(clientState, "valid test case %d failed: %s", i, tc.name)
-			suite.Require().True(clientState.IsFrozen(), "valid test case %d failed: %s", i, tc.name)
-			suite.Require().Equal(uint64(tc.evidence.GetHeight()), clientState.GetFrozenHeight(),
-				"valid test case %d failed: %s. Expected FrozenHeight %d got %d", tc.evidence.GetHeight(), clientState.GetFrozenHeight())
-		} else {
-			suite.Require().Error(err, "invalid test case %d passed: %s", i, tc.name)
-			suite.Require().Nil(clientState, "invalid test case %d passed: %s", i, tc.name)
-		}
+			if tc.expPass {
+				suite.Require().NoError(err, "valid test case %d failed: %s", i, tc.name)
+				suite.Require().NotNil(clientState, "valid test case %d failed: %s", i, tc.name)
```

Comment:
nit: when using `suite.Run` since the case name is passed in and printed out upon an error, I like to leave these error messages blank unless providing useful notes on why it might have failed